source/extensions/filters/http/cache/cache_filter.cc

```diff
@@ -126,6 +123,17 @@ Http::FilterDataStatus CacheFilter::encodeData(Buffer::Instance& data, bool end_
   return Http::FilterDataStatus::Continue;
 }
 
+void CacheFilter::onAboveWriteBufferHighWatermark() { ++high_watermark_calls_; }
+
+void CacheFilter::onBelowWriteBufferLowWatermark() {
+  ASSERT(high_watermark_calls_ > 0);
+  --high_watermark_calls_;
+  if (!remaining_ranges_.empty()) {
+    // Fetching the cached response body was stopped, continue if possible.
+    maybeGetBody();
+  }
+}
```

Comment:
Are there cases in which we stream through a router response and this might get triggered? I don't see any guards in maybeGetBody() to confirm that we are actually doing a fetch?